tools/zbdsniff

```diff
@@ -1,149 +1,110 @@
-        getnetworkkey(packet[1])
-        
-    cap.close()
-
-print "Processed %d capture files." % filecount
+from killerbee.scapy_extensions import *
+from killerbee.kbutils import *
+from zigbee_crypt import *
+from optparse import OptionParser
+
+ZBEE_SEC_ENC_MIC_32 = 0x5  #: ZigBee Seecurity Level id 5; Encrypted, 4 byte MIC
+
+APS_CMD = [
+    "Unknown",
+    "SKKE 1",
+    "SKKE 2",
+    "SKKE 3",
+    "SKKE 4",
+    "Transport Key",
+    "Update Device"
+]
+
+KEY_TYPE = [
+    "Trust Center",
+    "Network",
+    "Application Master",
+    "Aplication Link",
+    "Trust Link",
+    "High Security Network"
+]
+
+def sniffNetworkKey(pkts, key, verbose):
+    addrMap = dict()
+    keyHash = sec_key_hash(key, '\0')
+
+    for p in pkts:
+        if (p.haslayer("ZigbeeNWK")):
+            nwk = p.getlayer("ZigbeeNWK")
+            if nwk.source is not None and nwk.ext_src is not None:
+                if not nwk.source in addrMap and verbose:
+                    print("[+] Extended Source: " + ":".join("{:02x}".format(ord(ch)) for ch in raw(nwk)[8:16]) + " mapped to " + hex(nwk.source))
+                addrMap[nwk.source] = raw(nwk)[8:16]
+            if (p.haslayer("ZigbeeSecurityHeader") and p.haslayer("ZigbeeAppDataPayload")):
+                sec = p.getlayer("ZigbeeSecurityHeader")
+                if sec.key_type < 6:
+                    if nwk.source in addrMap:
+                        data = raw(p.getlayer("ZigbeeAppDataPayload"))
+                        scf = (ord(data[2]) & ~ ZBEE_SEC_ENC_MIC_32) | ZBEE_SEC_ENC_MIC_32
+                        a = data[0:2] + chr(scf) + data[3:7]
+                        c = data[7:-4]
+                        mic = data[-4:]
+                        nonce = addrMap[nwk.source] + data[3:7] + chr(scf)
+                        decrypted, success = decrypt_ccm(keyHash, nonce, mic, c, a)
+                        if success:
+                            print("[+] Decrypted:")
+                            if ord(decrypted[0]) < len(APS_CMD):
+                                print("    APS Command: " + APS_CMD[ord(decrypted[0])])
+                            if ord(decrypted[1]) < len(KEY_TYPE):
+                                print("    Key Type: " + KEY_TYPE[ord(decrypted[1])])
+                            print("    Value: " + ":".join("{:02x}".format(ord(ch)) for ch in decrypted[2:18]))
+                        else:
+                            print("[-] Decrypt failed - Wrong Key ???")
+                    else:
+                        print("[-] There was no ext_src mapping for: ", str(nwk.source))
+                else:
+                    print("[-] Unexpected key_type", str(sec.key_type))
+
+
+if __name__ == '__main__':
+    # Define the command line options.
+    parser = OptionParser(description="zbdsniff: Decode plaintext Zigbee Nwtwork key from a " +
+        "capture file. Will process libpcap or Daintree SNA capture files. Original concept: " +
```

Comment:
I think we'll now only support libpcap than also Daintree. Given the standard branching strategy, we would hold this for a major version release, 3.0. However, given the small usage of Daintree SNA and the other benefits of this new PR being much cleaner/better tested/etc -- and our disclaimers on the KillerBee project being active/etc -- I think this is fine to put into develop now.